octopuslite/utils.py

```diff
@@ -155,15 +157,16 @@ def parse_filename(filename: os.PathLike) -> dict:
 
     metadata = {
         "filename": filename,
-        "channel": Channels(int(params.group('channel'))),
-        "time": params.group('time'),
-        "position": params.group('position'),
-        "z": params.group('z'),
+        "channel": Channels(int(params.group("channel"))),
+        "time": params.group("time"),
+        "position": params.group("position"),
+        "z": params.group("z"),
         # "timestamp": os.stat(filename).st_mtime,
     }
 
     return metadata
 
+
 def image_generator(files, crop: Optional[tuple] = None):
```

Comment:
Also - I have to admit, I'm not totally sure why you need this function? Is it just if you don't want to use dask delayed?

Reply:
![image](https://user-images.githubusercontent.com/16838461/151353638-77173d87-6423-4ef0-b701-51db5ceb9275.png)

Yeah I've found it to be twice as quick in the localisation step. I just ran a quick test that doesn't seem like a massive difference but in my previous experience it's sometimes taken >5mins to run using dask and always around 30-40 seconds using an image generator. I also use the image generator function in the align notebook so figured instead of repeating myself I could incorporate it in octopus.

Reply:
try running the generator first and then the dask one - see if you get the same result

Reply:
it might be that you have some caching done (by the HDD itself) and so the second time you access the files, it's much quicker

Reply:
![image](https://user-images.githubusercontent.com/16838461/151361440-991f50ef-bc37-41ec-b790-c5fca7e79833.png)
maybe im splitting hairs but still a bit of difference in the timing

Reply:
that would be 2hrs comp time over my entire data set acquired since jan...

Reply:
in hindsight ive ended up removing this usage of the image generator from the localisation step. this is bc i moved the transformation application to as late as possible (after segmentation) so that we perturb as little of the data as possible with this translation. in doing this i need to use dask instead of image gen to load the images so i can apply transformation to them as well. however am still using the generator in the align notebook. could just take it out and have it as a function in that notebook if you would rather @quantumjot ??

Reply:
I think I would rather take it out of this